src/ape_ethereum/ecosystem.py

```diff
@@ -170,17 +134,66 @@ def raise_for_status(self):
             txn_hash = HexBytes(self.txn_hash).hex()
             raise TransactionError(message=f"Transaction '{txn_hash}' failed.")
 
+
+class BlockGasFee(BlockGasAPI):
     @classmethod
-    def decode(cls, data: dict) -> ReceiptAPI:
+    def decode(cls, data: Dict) -> BlockGasAPI:
+        return BlockGasFee(**data)  # type: ignore
+
+
+class BlockConsensus(BlockConsensusAPI):
+    @classmethod
+    def decode(cls, data: Dict) -> BlockConsensusAPI:
+        return cls(**data)  # type: ignore
+
+
+class Block(BlockAPI):
+    """
+    Class for representing a block on a chain.
+    """
+
+
+class Ethereum(EcosystemAPI):
+    @property
+    def config(self) -> EthereumConfig:
+        return self.config_manager.get_config("ethereum")  # type: ignore
+
+    def serialize_transaction(self, transaction: TransactionAPI) -> bytes:
+        if not transaction.signature:
+            raise SignatureError("The transaction is not signed.")
+
+        txn_data = transaction.as_dict()
+
+        # Don't publish from
+        if "from" in txn_data:
+            del txn_data["from"]
+
+        unsigned_txn = serializable_unsigned_transaction_from_dict(txn_data)
+        signature = (
+            transaction.signature.v,
+            to_int(transaction.signature.r),
+            to_int(transaction.signature.s),
+        )
+
+        signed_txn = encode_transaction(unsigned_txn, signature)
+
+        if transaction.sender and EthAccount.recover_transaction(signed_txn) != transaction.sender:
+            raise SignatureError("Recovered Signer doesn't match sender!")
+
+        return signed_txn
+
+    def decode_receipt(self, data: dict) -> ReceiptAPI:
         status = data.get("status")
+
         if status:
             if isinstance(status, str) and status.isnumeric():
                 status = int(status)
 
             status = TransactionStatusEnum(status)
 
         txn_hash = data["hash"].hex() if isinstance(data["hash"], HexBytes) else data["hash"]
-        return cls(  # type: ignore
+
+        return Receipt(  # type: ignore
```

Comment:
To get this to work with `ape-geth`, I had to change line 208 to `receiver=data["to"] or ""`

What was happening was that the data was `{"receiver": None}` for geth and  `{"receiver": ""}` for eth-tester. The empty str passes the Pydantic validation and the `None` does not.